### PR TITLE
Merges from develop/1.14

### PR DESCRIPTION
**Pull request by byrnHDF**
Fix release action that allows files to be uploaded.
Add autotools h5copy help test.
Update cmake tools.cmake file for clang-tidy.
CMake build is missing H5FDsubfiling.h
macros improved for configure output capture.
Remove obsolete files

### CODE REVIEW

File: config/cmake/HDFLibMacros.cmake [hunk 1]
```diff
@@ -17,13 +17,14 @@ macro (ORIGINAL_ZLIB_LIBRARY compress_type)
...
         URL ${ZLIB_URL}
         URL_HASH ""
     )
   endif ()
   FetchContent_GetProperties(HDF5_ZLIB)
-  if(NOT zlib_POPULATED)
+  if(NOT hdf5_zlib_POPULATED)
     FetchContent_Populate(HDF5_ZLIB)
 
     # Copy an additional/replacement files into the populated source
```
Comment (lrknox) [inline config/cmake/HDFLibMacros.cmake:30]:
```suggestion
    # Copy an additional/replacement file into the populated source
```

File: release_docs/RELEASE.txt [hunk 2]
```diff
@@ -409,7 +409,15 @@ Bug Fixes since HDF5-1.12.2 release
 
     Java Library
     ------------
-    -
+    - Fixed switch case 'L' block missing a break statement.
+
+      The HDF5Array.arrayify method is missing a break statement in the case 'L': section
```
Comment (lrknox) [inline release_docs/RELEASE.txt:414]: This should probably be "was missing" and "caused" to agree with "Fixed" and "was fixed".

File: src/H5Rpublic.h [hunk 3]
```diff
@@ -75,7 +75,7 @@ typedef haddr_t hobj_ref_t;
 /**
  * Buffer to store heap ID and index
  *
- * This needs to be large enough to store largest #haddr_t in a worst case
+ * This needs to be large enough to store largest haddr_t in a worst case
```
Comment (lrknox) [inline src/H5Rpublic.h:78]: large enough to store the largest possible haddr_t?

What's a worst case machine?

**Review by lrknox — APPROVED** [review]

File: config/cmake/HDFLibMacros.cmake (hunk 1 above)
Comment (byrnHDF) [reply config/cmake/HDFLibMacros.cmake:30]: This will need to be fixed in develop, then 1.14 first.

File: release_docs/RELEASE.txt (hunk 2 above)
Comment (byrnHDF) [reply release_docs/RELEASE.txt:414]: Another item for the "fix the documentation" issue to fix in all branches.

File: src/H5Rpublic.h (hunk 3 above)
Comment (byrnHDF) [reply src/H5Rpublic.h:78]: This change was only to fix doxygen warning.
Another item for the "fix the documentation" issue to fix in all branches.

**Pull request merged**